File: src/main/java/com/github/aivancioglo/resttest/setters/OAuth2Token.java
package com.github.aivancioglo.resttest.setters;

import com.github.aivancioglo.resttest.http.HTTPRequest;

public class OAuth2Token<T extends HTTPRequest> implements Setter<T> {
    private String token;

    public OAuth2Token(String token) {
        this.token = token;
    }

    /**
     * Update request.
     * @param request actual request.
     */
    @Override
    public void update(T request) {
        request.setOAuth2Token(token);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/setters/OAuth2Token.kt
package com.github.aivancioglo.resttest.setters

import com.github.aivancioglo.resttest.http.HTTPRequest

class OAuth2Token<in T : HTTPRequest<*>>(val token: String) : Setter<T> {

    /**
     * Setting OAuth 2 token to your request.
     *
     * @param request that will be updated.
     */
    override fun update(request: T) {
        request.oAuth2.token = token
    }
}